### DISCUSSION TITLE
Excessive memory use?

### DISCUSSION THREAD

**Discussion by ghost** [jcwillox/hass-auto-backup#172] [Q&A]
I've noticed that every time I run a full backup the memory usage for HAOS in my Proxmox VM increases by appx 1GB. My VM is 12GB, so a daily backup will max out within just a few days. Coincidentally, each zipped backup file is also appx 1GB is size. 

HA starts to get quite sluggish once it nears 12GB then I have to reboot the VM.

How can I free this memory without rebooting?

In the attached image you can see I ran a backup twice.

![1735116779724](https://github.com/user-attachments/assets/de0be52a-33a9-4786-a8a9-1e409c2b71a3)

**Comment by jcwillox** [comment]
This would be a question for the Home Assistant team / community as its unrelated to Auto Backup, as the Supervisor (Home Assistant) is the one doing the actual backup.

It sounds like a memory leak as the memory usage should go back down after the backup completes, and 1gb of memory to compress an archive sounds like a lot on its own. Unfortunately I don't know why or how to fix that, I'd recommend bringing it up with Home Assistant.

**Reply by ghost** [reply]
Thanks. I just started using the new built-in HA backup function in 2024.12.1, it's only been a couple days but there are no memory leaks so far. So I will guess the 'solution' is to just stop using this integration.

**Reply by jcwillox** [reply]
Well that's good to hear, though there's practically no way this integration could cause that, as all it does it tell HA to make a backup, and then stores the backup id and when to delete it. So either what's included in the backup is different or they fixed the issue in the update.
